pkg/snapstore/abs_snapstore_test.go

```diff
@@ -0,0 +1,301 @@
+		Segment: blobFlatListSegment{
+			BlobItems: blobs,
+		},
+		MaxResults: int32(limit),
+		NextMarker: nextMaker,
+	}
+
+	rawXML, err := xml.MarshalIndent(listBlobsFlatSegmentResponse, "", "")
+	if err != nil {
+		return err
+	}
+
+	w.Body = ioutil.NopCloser(strings.NewReader(xml.Header + string(rawXML)))
+	w.StatusCode = http.StatusOK
+	return nil
+}
+
+// handleBlobCreateOperation responds with a blob `Put` response.
+func (p *fakePolicy) handleBlobPutOperation(w *http.Response) {
+	var (
+		query   = w.Request.URL.Query()
+		comp    = query.Get("comp")
+		blockid = query.Get("blockid")
+		key     = parseObjectNamefromURL(w.Request.URL)
+	)
+
+	switch comp {
+	case "block":
+		content, err := ioutil.ReadAll(w.Request.Body)
+		if err != nil {
+			w.StatusCode = http.StatusBadRequest
+			w.Body = ioutil.NopCloser(strings.NewReader(fmt.Sprintf("failed to read content %v", err)))
+			return
+		}
+
+		p.multiPartUploadsMutex.Lock()
+		blockList, ok := p.multiPartUploads[key]
+		if !ok {
+			blockList = make(map[string][]byte, 0)
+		}
+		blockList[blockid] = content
+		p.multiPartUploads[key] = blockList
+		p.multiPartUploadsMutex.Unlock()
+		w.StatusCode = http.StatusCreated
+
+	case "blocklist":
+		content, err := ioutil.ReadAll(w.Request.Body)
+		if err != nil {
+			w.StatusCode = http.StatusBadRequest
+			w.Body = ioutil.NopCloser(strings.NewReader(fmt.Sprintf("failed to read content %v", err)))
+			return
+		}
+		blockLookupXML := strings.TrimPrefix(string(content), xml.Header)
+		var blockLookupList azblob.BlockLookupList
+		if err := xml.Unmarshal([]byte(blockLookupXML), &blockLookupList); err != nil {
+			w.StatusCode = http.StatusBadRequest
+			w.Body = ioutil.NopCloser(strings.NewReader(fmt.Sprintf("failed to parse body %v", err)))
+			return
+		}
+		blockContentMap := p.multiPartUploads[key]
+		content = make([]byte, 0)
+		for _, blockID := range blockLookupList.Latest {
+			content = append(content, blockContentMap[blockID]...)
+		}
+		p.objectMap[key] = &content
+		w.StatusCode = http.StatusCreated
+	}
+	w.Body = http.NoBody
+}
+
+// handleBlobGetOperation on GET request `/testContainer/testObject` responds with a `Get` response.
+func (p *fakePolicy) handleBlobGetOperation(w *http.Response) {
+	key := parseObjectNamefromURL(w.Request.URL)
+	if _, ok := p.objectMap[key]; ok {
+		w.StatusCode = http.StatusOK
+		w.Body = ioutil.NopCloser(bytes.NewReader(*p.objectMap[key]))
+	} else {
+		w.StatusCode = http.StatusNotFound
+		w.Body = http.NoBody
+	}
+}
+
+// handleDeleteObject on delete request `/testContainer/testObject` responds with a `Delete` response.
+func (p *fakePolicy) handleDeleteObject(w *http.Response) {
+	key := parseObjectNamefromURL(w.Request.URL)
+	if _, ok := p.objectMap[key]; ok {
+		delete(p.objectMap, key)
+		w.StatusCode = http.StatusAccepted
+	} else {
+		w.StatusCode = http.StatusNotFound
+	}
+	w.Body = http.NoBody
+}
+
+/////////////////////////////////////////////
+// Truncated azblob types for XML encoding //
```

Comment:
Are these comments required? Including the ones on lines 278 and 298.

Reply:
Yes. I think, the short comment is enough to explain why we need to create these types.